master/static/srv/check_ready_logs.py

```diff
@@ -24,22 +24,25 @@
     RequestException,
     giveup=lambda e: e.response is not None and e.response.status_code < 500,
 )
-def post_ready(master_url: str, cert: certs.Cert, allocation_id: str):
+def post_ready(master_url: str, cert: certs.Cert, allocation_id: str, state: str):
     api.post(
         master_url,
-        f"/api/v1/allocations/{allocation_id}/ready",
+        f"/api/v1/allocations/{allocation_id}/{state}",
         {},
         cert=cert,
     )
 
 
-def main(ready: Pattern):
+def main(ready: Pattern, waiting: Optional[Pattern] = None):
     master_url = str(os.environ["DET_MASTER"])
     cert = certs.default_load(master_url)
     allocation_id = str(os.environ["DET_ALLOCATION_ID"])
     for line in sys.stdin:
         if ready.match(line):
-            post_ready(master_url, cert, allocation_id)
+            post_ready(master_url, cert, allocation_id, "ready")
+            return
+        if waiting and waiting.match(line):
+            post_ready(master_url, cert, allocation_id, "waiting")
```

Comment:
random drive by comment, why do we return, shouldn't we be sending both ready and waiting for tb?

Reply:
Great catch! Thank you!